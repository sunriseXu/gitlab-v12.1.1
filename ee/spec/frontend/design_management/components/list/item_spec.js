import { shallowMount } from '@vue/test-utils';
import Item from 'ee/design_management/components/list/item.vue';

describe('Design management list item component', () => {
  let vm;

  function createComponent(commentsCount = 1) {
    vm = shallowMount(Item, {
      propsData: {
        id: 1,
        name: 'test',
        image: 'http://via.placeholder.com/300',
        commentsCount,
        updatedAt: '01-01-2019',
      },
      stubs: ['router-link'],
    });
  }

  afterEach(() => {
    vm.destroy();
  });

  it('renders item with single comment', () => {
    createComponent();

    expect(vm.element).toMatchSnapshot();
  });

  it('renders item with multiple comments', () => {
    createComponent(2);

    expect(vm.element).toMatchSnapshot();
  });

  it('hides comment count', () => {
    createComponent(0);

    expect(vm.element).toMatchSnapshot();
  });
});
