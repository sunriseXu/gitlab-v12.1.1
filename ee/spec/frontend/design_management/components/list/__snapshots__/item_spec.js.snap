// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management list item component hides comment count 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0"
  >
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 design-img"
      height="230"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component renders item with multiple comments 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0"
  >
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 design-img"
      height="230"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <div
      class="ml-auto d-flex align-items-center text-secondary"
    >
      <icon-stub
        class="ml-1"
        cssclasses=""
        name="comments"
        size="16"
      />
       
      <span
        aria-label="2 comments"
        class="ml-1"
      >
        
        2
      
      </span>
    </div>
  </div>
</router-link-stub>
`;

exports[`Design management list item component renders item with single comment 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0"
  >
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 design-img"
      height="230"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <div
      class="ml-auto d-flex align-items-center text-secondary"
    >
      <icon-stub
        class="ml-1"
        cssclasses=""
        name="comments"
        size="16"
      />
       
      <span
        aria-label="1 comment"
        class="ml-1"
      >
        
        1
      
      </span>
    </div>
  </div>
</router-link-stub>
`;
